### PR TITLE
MRG: Mixin time_as_index

### LINKED ISSUE #3083: Add two methods Epochs, Evoked
It would be super useful to have these methods added to the Epochs, and Evoked.

`raw.time_as_index` and `raw.index_as_time`

### PR DESCRIPTION
**Pull request by Eric89GXL**
Closes #3083.

Ready for review/merge from my end.

### CODE REVIEW

File: mne/io/base.py [hunk 1]
```diff
@@ -202,6 +202,50 @@ def to_data_frame(self, picks=None, index=None, scale_time=1e3,
         return df
 
 
+class TimeMixin(object):
+    """Class to add sfreq and time_as_index capabilities to certain classes."""
+
+    def time_as_index(self, times, use_first_samp=None, use_rounding=False):
```
Comment (agramfort) [inline mne/io/base.py]: I don't really like the fact that a use_first_samp param is exposed for non raw objects.

File: mne/io/base.py [hunk 2]
```diff
@@ -202,6 +202,50 @@ def to_data_frame(self, picks=None, index=None, scale_time=1e3,
...
+        """Convert time to indices
+
+        Parameters
+        ----------
+        times : list-like | float | int
+            List of numbers or a number representing points in time.
+        use_first_samp : boolean
+            This is deprecated and will be removed in 0.13.
+            If True, time is treated as relative to the session onset, else
+            as relative to the recording onset. Default is False.
```
Comment (agramfort) [inline mne/io/base.py]: even more as it's deprecated

File: mne/io/base.py (hunk 1 above)
Comment (Eric89GXL) [inline mne/io/base.py]: Okay, I can put one a different one in for `_BaseRaw` for one release, then

File: mne/io/cnt/tests/test_cnt.py [hunk 3]
```diff
@@ -4,23 +4,29 @@
...
 
 @testing.requires_testing_data
 def test_data():
     """Test reading raw cnt files."""
-    raw = _test_raw_reader(read_raw_cnt, montage=None, input_fname=fname,
-                           eog='auto', misc=['NA1', 'LEFT_EAR'])
+    with warnings.catch_warnings(record=True) as w:
+        raw = _test_raw_reader(read_raw_cnt, montage=None, input_fname=fname,
+                               eog='auto', misc=['NA1', 'LEFT_EAR'])
+    assert_true(all('meas date' in str(ww.message) for ww in w))
```
Comment (Eric89GXL) [inline mne/io/cnt/tests/test_cnt.py]: cc @jaeilepp we should catch all warnings during tests

**Comment by Eric89GXL** [comment]
@agramfort done

**Comment by agramfort** [comment]
thanks @Eric89GXL

**Pull request merged**